File: bidly-backend/src/main/java/com/bidly/bidly_backend/model/SubastaSesion.java
package com.bidly.bidly_backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "subasta_sesion")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SubastaSesion {

    @Id
    @Column(name = "subasta")
    private Long subasta;

    @Column(name = "item_activo", nullable = false)
    private Long itemActivo;

    @Column(name = "orden_actual", nullable = false)
    private Integer ordenActual = 1;

    @Column(name = "timer_desde", nullable = false)
    private LocalDateTime timerDesde;

    @Column(name = "iniciada_en", nullable = false)
    private LocalDateTime iniciadaEn;
}

File: bidly-fastapi/app/models/subasta_sesion.py
from sqlalchemy import Column, Integer, DateTime, ForeignKey
from app.database import Base


class SubastaSesion(Base):
    __tablename__ = "subasta_sesion"

    subasta      = Column(Integer, ForeignKey("subastas.identificador"), primary_key=True)
    item_activo  = Column(Integer, ForeignKey("itemscatalogo.identificador"))
    orden_actual = Column(Integer)
    timer_desde  = Column(DateTime)
    iniciada_en  = Column(DateTime)
